content/blog/webr-0-2-0/index.Rmd

```diff
@@ -0,0 +1,535 @@
+
+Today, on the other hand, `OffscreenCanvas` is [supported](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas#browser_compatibility) in all major desktop and mobile browsers. Safari has supported it since version 16.4, and so with webR 0.2.0 we have rewritten the [`webr::canvas()`](https://docs.r-wasm.org/webr/latest/api/r.html#graphics-device-for-drawing-to-a-html-canvas-element) graphics device to take full advantage of the `OffscreenCanvas` interface. This has led to a significant performance improvement, particularly when creating plots containing many points. The two videos below show the same plot rendered in webR 0.1.1 and 0.2.0, the difference is not just visible, but an order of magnitude faster.
+
+<video controls loop width="100%" src="plot.mp4" style="border: 2px solid #CCC;">
+
+<source src="plot.mp4">
+
+</video>
+
+::: {style="text-align: center; font-weight: bold;"}
+A performance comparison plotting 300000 points in webR 0.1.1 and 0.2.0.
+:::
+
+A potential downside is that users of less up-to-date browsers without `OffscreenCanvas` support won't be able to use the [`webr::canvas()`](https://docs.r-wasm.org/webr/latest/api/r.html#graphics-device-for-drawing-to-a-html-canvas-element) graphics device. Such users should instead make use of our additional updates to webR to support the traditional Cairo-based bitmap devices. The [built-in graphics devices section](#built-in-bitmap-graphics-devices) discusses that in more detail.
+
+### Modern text rendering and internationalisation
+
+With webR 0.1.1, the canvas graphics device had only minimal support for rendering text. The typeface was fixed, the font metrics were estimated with a heuristic, and Unicode characters outside the Basic Latin block often failed to render. It worked most of the time, but it was far from ideal. This area of software engineering is [suprisingly difficult](https://faultlore.com/blah/text-hates-you/) to get right, and even native installations of R can have [serious text rendering issues](https://www.tidyverse.org/blog/2021/02/modern-text-features/).
+
+In comparison, web browser support for text rendering is excellent. Now that we use the `OffscreenCanvas` interface, we too can take advantage of the years of work behind browser's support for text on the web. The example below demonstrates several of the modern text rendering features now supported by [`webr::canvas()`](https://docs.r-wasm.org/webr/latest/api/r.html#graphics-device-for-drawing-to-a-html-canvas-element).
+
+```{webr}
+plot(rnorm(1000), rnorm(1000), col=rgb(0,0,0,0.5),
+     xlim=c(-5, 5), ylim=c(-5, 5), main="This is the title 🚀",
+     xlab="This is the x label", ylab="This is the y label",
+     family="Futura")
+text(-3.5, 4, "This is English", family="monospace")
+text(-3.5, -4, "هذا مكتوب باللغة العربية")
+text(3.5, 4, "これは日本語です")
+text(3.5, -4, "זה כתוב בעברית")
+```
```

Comment:
In code blocks, I recommend spaces around `=` to conform to the Tidyverse style guide. I would also recommend a line break after the first `(`, something like

```r
plot(
  rnorm(1000), rnorm(1000), 
  col = rgb(0,0,0,0.5),
  xlim = c(-5, 5), ylim = c(-5, 5), 
  main = "This is the title 🚀",
  xlab = "This is the x label", 
  ylab = "This is the y label",
  family = "Futura"
)
text(-3.5, 4, "This is English", family="monospace")
text(-3.5, -4, "هذا مكتوب باللغة العربية")
text(3.5, 4, "これは日本語です")
text(3.5, -4, "זה כתוב בעברית")
```